homeassistant/components/template/config.py

```diff
@@ -70,14 +71,39 @@
     vacuum as vacuum_platform,
     weather as weather_platform,
 )
-from .const import DOMAIN, PLATFORMS, TemplateConfig
+from .const import CONF_DEFAULT_ENTITY_ID, DOMAIN, PLATFORMS, TemplateConfig
 from .helpers import async_get_blueprints, rewrite_legacy_to_modern_configs
 
 _LOGGER = logging.getLogger(__name__)
 
 PACKAGE_MERGE_HINT = "list"
 
 
+def validate_binary_sensor_auto_off_has_trigger(obj: dict) -> dict:
+    """Validate that binary sensors with auto_off have triggers."""
+    if CONF_TRIGGERS not in obj and DOMAIN_BINARY_SENSOR in obj:
+        binary_sensors: list[ConfigType] = obj[DOMAIN_BINARY_SENSOR]
+        for binary_sensor in binary_sensors:
+            if binary_sensor_platform.CONF_AUTO_OFF in binary_sensor:
```

Comment:
We could invert some of these conditions and use `continue` and similar to keep the code less indented.